### PR TITLE
Fix CA0114, CS0108, CS0618, CS1572, CS1573

### PR DESCRIPTION
**Pull request by azabbasi**
https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0114
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0108
https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1572
https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1573
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0618

### CODE REVIEW

File: e2e/test/iothub/DeviceTokenRefreshE2ETests.cs [hunk 1]
```diff
@@ -198,16 +198,6 @@ private async Task DeviceClient_TokenIsRefreshed_Internal(Client.TransportType t
             }
         }
 
-        public void Dispose()
```
Comment (azabbasi) [inline e2e/test/iothub/DeviceTokenRefreshE2ETests.cs:201]: already exists in the base class. nothing new to dispose

File: e2e/test/iothub/FileUploadE2ETests.cs [hunk 2]
```diff
@@ -25,6 +25,7 @@ public class FileUploadE2ETests : E2EMsTestBase
 
         [LoggedTestMethod]
         [TestCategory("LongRunning")]
+        [Obsolete]
```
Comment (azabbasi) [inline e2e/test/iothub/FileUploadE2ETests.cs:28]: Test hits obsolete class members so it has to be marked as obsolete also.

File: iothub/service/tests/JobClient/HttpJobClientTests.cs [hunk 3]
```diff
@@ -23,7 +23,6 @@ public class HttpJobClientTests
         private readonly TimeSpan timeout = TimeSpan.FromMinutes(1);
 
         private Mock<IHttpClientHelper> httpClientHelperMock;
-        private readonly string iotHubName = "test";
```
Comment (azabbasi) [inline iothub/service/tests/JobClient/HttpJobClientTests.cs:26]: unused.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**